azure-mgmt-sql/src/main/java/com/microsoft/azure/management/sql/SqlDatabase.java

```diff
@@ -314,6 +330,22 @@
         }
 
         /**
+         * The SQL Database definition to set the Max Size in Bytes for database.
+         */
+        interface WithMaxSizeBytes {
+            /**
+             * Sets the max size in bytes for SQL Database.
+             *
+             * @param maxSizeBytes max size of the Azure SQL Database expressed in bytes. Note: Only
+             * the following sizes are supported (in addition to limitations being
+             * placed on each edition): { 100 MB | 500 MB |1 GB | 5 GB | 10 GB | 20
+             * GB | 30 GB … 150 GB | 200 GB … 500 GB }
+             * @return The next stage of the definition.
+             */
+            WithCreateWithElasticPoolOptions withMaxSizeBytes(long maxSizeBytes);
```

Comment:
May be in future, in these type of scenarios, where user needs to do some math for conversion (here nGB to equivalent  bytes, mMB to equivalent bytes etc), we can think of having an overload like:

```java
withMaxSizeBytes(long size, SizeUnit unit)

where SizeUnit is an enum { GB, MB, KB, Byte }.
```

See similar example here    http://reactivex.io/RxJava/javadoc/rx/Observable.html#delay(long,%20java.util.concurrent.TimeUnit)

```java
Observable<T> delay(long delay, java.util.concurrent.TimeUnit unit)
```